### ISSUE TITLE
REPL: Control-C behavior

### ISSUE THREAD

**Issue by cb22** [tigerbeetle/tigerbeetle#1186]
When hitting Control-C inside the REPL, it exits immediately, even if you're in the middle of typing or a command. Rather, if you're midway through a command, it should clear whatever you've entered, and if you're at a blank prompt, only then should it exit the REPL (maybe after twice in a row, like Node does).

**Comment by alxgh** [comment]
https://github.com/tigerbeetle/tigerbeetle/pull/1255

**Comment by swaingotnochill** [comment]
Hi, I would like to pick this up. Can you assign me?
I will triage and see what's the issue on my end.

**Comment by fidelicura** [comment]
Looks like this issue is fixed, see https://github.com/tigerbeetle/tigerbeetle/pull/1255#issuecomment-2566379581

**Comment by ryukyi** [comment]
> even if you're in the middle of typing or a command

This part is fixed - it clears and jumps to new line

> if you're at a blank prompt, only then should it exit the REPL (maybe after twice in a row, like Node does).

Ctrl+C doesn't exit the repl on blank prompts:

<img width="584" height="704" alt="Image" src="https://github.com/user-attachments/assets/20f0652c-bbcd-4da7-93ff-11e8ac9a72b4" />

**Comment by matklad** [comment]
Yeah, this works as intended, Ctrl-C clears input, Ctrl-D exits Repl, following conventions of `python`
